### DISCUSSION TITLE
Github secrets defined in forked repo are not accessible in same repo

### DISCUSSION THREAD

**Discussion by vbmade2000** [community/community#26283] [Actions]
<p>Hi experts,</p>
<p>I have a forked repo. I am trying to create a Github workflow that creates a snap package and upload it. I use <strong>snapcore/action-publish</strong> action to publish my snap. It requires env var <strong>SNAPCRAFT_STORE_CREDENTIALS</strong> which I have stored as repository secret in my repo. But the issue is because secret is not accessible, publish steps fails. Please make note that I have added secrets in forked repo and not in upstream as I know that secrets from upstream repo is not available in fork.</p>
<p>Here is a yaml.</p>
<pre><code class="lang-auto">name: Snap

on:
  push:
    tags:
      - "*"

jobs:
  create-snap:
    name: Create snap
    runs-on: ubuntu-latest
    steps:
      - name: Checkout source
        uses: actions/checkout@v3
      - name: Build snap
        uses: snapcore/action-build@v1
        id: build-snap
      - name: Print secret
        run: |
            echo "This is secret"
            echo ${{ secrets.MAC_NAME }}
      - name: Publish snap
        uses: snapcore/action-publish@v1
        env:
          SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.STORE_LOGIN }}
        with:
          snap: ${{ steps.build-snap.outputs.snap }}
          release: edge
</code></pre>
<p><strong>echo ${{ secrets.MAC_NAME }}</strong> is also not printing anything.</p>
<p>Please help.</p>

**Comment by vbmade2000** [comment, marked as answer]
<p>It is solved now. It required clone and push using SSH keys instead of regular username and personal token.</p>

**Comment by Akash1134** [comment]
<p>Thanks for coming back and updating the discussion. :slight_smile: This might help others with similar query.</p>
